### PR TITLE
ENH: Tedana-friendly ME derivatives

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

* Add `--output-echos` flag to indicate that BOLD series should be output for individual echos.

TODO:

* [x] Figure out what's going on with SDC. Currently all the logic is in a branch that assumes no SDC.
* [x] ...

Beginning to address #2542.
Builds on #2574.

### CODE REVIEW

**Comment by effigies** [comment]
Need to update the expected derivatives for ds000210 and do a style pass, but otherwise I think this is ready for review. Definitely did some quick-and-dirty patches to get all of the metadata in, which could stand some thoughtful cleaning up.

File: fmriprep/cli/parser.py [hunk 1]
```diff
@@ -315,6 +315,14 @@ def _slice_time_ref(value, parser):
 https://fmriprep.readthedocs.io/en/%s/spaces.html"""
         % (currentv.base_version if is_release else "latest"),
     )
+    g_conf.add_argument(
+        "--output-echos",
```
Comment (oesteban) [inline fmriprep/cli/parser.py:319]:
```suggestion
        "--me-output-echos",
```

Signal the subtype of workflow like the `--fs-*`

File: fmriprep/config.py [hunk 2]
```diff
@@ -410,6 +410,8 @@ class execution(_Config):
     """Do not monitor *fMRIPrep* using Sentry.io."""
     output_dir = None
     """Folder where derivatives will be stored."""
+    output_echos = False
```
Comment (oesteban) [inline fmriprep/config.py:413]:
```suggestion
    me_output_echos = False
```

File: fmriprep/workflows/bold/outputs.py [hunk 3]
```diff
@@ -240,6 +244,22 @@ def init_func_derivatives_wf(
             (raw_sources, ds_bold_mask_native, [('out', 'RawSources')]),
         ])
 
+    if config.execution.output_echos and multiecho:
```
Comment (oesteban) [inline fmriprep/workflows/bold/outputs.py:247]:
```suggestion
    if multiecho and config.execution.me_output_echos:
```

File: fmriprep/workflows/bold/base.py [hunk 4]
```diff
@@ -687,17 +703,6 @@ def init_func_preproc_wf(bold_file, has_fieldmap=False):
         ])
         # fmt:on
 
-    if nonstd_spaces.intersection(("func", "run", "bold", "boldref", "sbref")):
```
Comment (oesteban) [inline fmriprep/workflows/bold/base.py:690]: does this mean that the boldref is now generated always? (fine by me, if so)

**Review by oesteban — APPROVED** [review]

File: fmriprep/workflows/bold/base.py (hunk 4 above)
Comment (effigies) [reply fmriprep/workflows/bold/base.py:690]: No, this just ensures these inputs are always available to `func_derivatives_wf`. You still need to enable the output space to create the datasink nodes.

**Comment by effigies** [comment]
This is ready and can either go in 21.0.0 or wait for the next series.

**Comment by oesteban** [comment]
My vote goes to 21.0.0, FWIW 👍

**Pull request merged**